mocks_test.go

```diff
@@ -179,13 +179,17 @@ func (l *mockLogger) Logf(message string, args ...interface{}) {
 type mockReporter struct {
 	testing  *testing.T
 	reported bool
+	reportCb func()
 }
 
 func newMockReporter(t *testing.T) *mockReporter {
 	return &mockReporter{testing: t}
 }
 
 func (r *mockReporter) Errorf(message string, args ...interface{}) {
+	if r.reportCb != nil {
+		r.reportCb()
+	}
 	r.testing.Logf("Fail: "+message, args...)
 	r.reported = true
 }
```

Comment:
Nitpick: let's set reported first and then call reportCb. So that when callback is called, it will observe that reported is already true.